impeller/renderer/backend/vulkan/command_buffer_vk.cc

```diff
@@ -55,13 +55,24 @@ bool CommandBufferVK::OnSubmitCommands(CompletionCallback callback) {
   if (!encoder_) {
     encoder_ = encoder_factory_->Create();
   }
-  if (!callback) {
-    return encoder_->Submit();
+  auto context = context_.lock();
```

Comment:
The problem with this approach is that I don't really know if this is being created during a frame workload or not. If this is an image decoding command buffer, or something being used for a layout transition on Vulkan, then it won't get submitted until we render a frame.

Maybe that is OK, so we won't technically queue up iO thread work until its consumed by a frame, but so what?

Reply:
Well, besides the problem with synchronization I guess.